File: ScaleformUI_Lua/src/base/ItemFont.lua
ItemFont = setmetatable({}, ItemFont)

function ItemFont.New(fontName, fontId)
    if fontId == nil then fontId = 0 end
    local font = {
        FontName = fontName,
        FontID = fontId
    }
    return setmetatable(font, ItemFont)
end

function ItemFont.RegisterFont(gfxName, fontName)
    RegisterFontFile(gfxName)
    local font = {
        FontName = fontName,
        FontID = RegisterFontId(fontName)
    }
    return setmetatable(font, ItemFont)
end

File: ScaleformUI_V8/src/elements/item-font.ts
export class ItemFont {
    constructor(public readonly fontName: string, public readonly fontId = 0) {}

    public static registerFont(gfxName: string, fontName: string) {
        RegisterFontFile(gfxName)
        return new ItemFont(fontName, RegisterFontId(fontName))
    }
}